pkg/kubelet/container/helpers.go

```diff
@@ -145,6 +146,19 @@ func ExpandContainerCommandOnlyStatic(containerCommand []string, envs []v1.EnvVa
 	return command
 }
 
+func ExpandContainerVolumeMounts(container *v1.Container, envs []EnvVar) (volumeMounts []v1.VolumeMount) {
+	mapping := expansion.MappingFuncFor(EnvVarsToMap(envs))
+
+	if len(container.VolumeMounts) != 0 {
+		for _, mount := range container.VolumeMounts {
+			mount.SubPath = expansion.Expand(mount.SubPath, mapping)
```

Comment:
Subpath name generation should be done in kubelet_pods.go right before we do subpath processing and validation.

Reply:
There also needs to be a feature gate around this.

Reply:
I think this is a better place for the subpath name generation:
https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/kubelet_pods.go#L168

Reply:
Feature gates are added in this file:
https://github.com/kubernetes/kubernetes/blob/master/pkg/features/kube_features.go

And you use them to gate your code like this: 
https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/kubelet_pods.go#L165

Reply:
@msau42 I have been looking at moving the Expansion into the makeMounts function and I think this is too late in the processing. The subpath expansion is in the container spec along with the expansion of command args and needs to be performed before it gets to the subpath mounting section. - it is not really a subpath function, it is a container function. Do you agree?

Reply:
I'm not sure why it is too late?  `makeMounts` is called in `GenerateRunContainerOptions` which is currently called right after this is called.  So this is not changing the order, it's just moving the subpath name generation closer to the rest of subpath processing and validation.

I mainly want to avoid a scenario in the future where someone may try to change the ordering, and may not realize there's an ordering dependency because the makeMounts function is buried under another call.

Reply:
@msau42 Okay. Thanks. I'll take another look at it